qng/common.py

```diff
@@ -0,0 +1,69 @@
+import argparse
+import json
+import operator
+import os
+import random
+import sys
+import unicodedata
+import re
+from typing import Optional
+
+import qng
+
+
+_BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+_DATA_DIR = os.path.join(_BASE_DIR, 'data')
+
+
+def _validate_snake_kebab_args(args):
+    if args.kebab_case and args.snake_case:
+        print('Cannot specify --snake-case and --kebab-case.', file=sys.stderr)
+        sys.exit(1)
+
+
+def _strip_diacritics(string: str) -> str:
+    return (
+        unicodedata.normalize('NFKD', string)
+        .encode('ascii', 'ignore')
+        .decode('utf-8')
+    )
+
+
+def _snakebabify_name(name: str, sep: str) -> str:
+    name = _strip_diacritics(name)
+    name = name.lower()
+    name = re.sub(r'[^a-zA-Z0-9_]', sep, name)
```

Comment:
Just for the record, regarding your commit message on this, the original intent behind the `--snake-case` wasn't directly to produce valid C identifiers, since my intended use case was for Docker container names, where something like `marie-chantale_tremblay` is perfectly valid, and perhaps even more legible than `marie_chantale_tremblay`.

I think I'd your snake case changes as-is so there's the option of producing valid C idents, but then reintroduce the old format as `--docker-case` or some other name perhaps for my own use case.